src/uct/tcp/tcp_cm.c

```diff
@@ -167,16 +159,27 @@ ucs_status_t uct_tcp_cm_send_event(uct_tcp_ep_t *ep, uct_tcp_cm_conn_event_t eve
                 (ep->conn_state != UCT_TCP_EP_CONN_STATE_CONNECTED),
                 "ep=%p", ep);
 
-    pkt_length        = sizeof(*pkt_hdr);
+    pkt_length                  = sizeof(*pkt_hdr);
     if (event == UCT_TCP_CM_CONN_REQ) {
-        cm_pkt_length = sizeof(*conn_pkt);
+        cm_pkt_length           = sizeof(*conn_pkt);
+
+        if (ep->conn_state == UCT_TCP_EP_CONN_STATE_CONNECTING) {
+            magic_number_length = sizeof(uct_tcp_cm_magic_number);
+        }
     } else {
-        cm_pkt_length = sizeof(event);
+        cm_pkt_length           = sizeof(event);
+    }
+
+    pkt_length += cm_pkt_length + magic_number_length;
+    pkt_buf     = ucs_alloca(pkt_length);
+
+    if ((event == UCT_TCP_CM_CONN_REQ) &&
```

Comment:
maybe put this inside lines 187-189? to check only UCT_TCP_EP_CONN_STATE_CONNECTING